platform/core/src/feature_categories.rs

```diff
@@ -0,0 +1,581 @@
+                    feature_manifest.thumbnail.unwrap()
+                )
+            )
+        );
+        assert_eq!(feature.thumbnail_color, feature_manifest.thumbnail_color.unwrap());
+        assert_eq!(feature.primary_color, feature_manifest.primary_color.unwrap());
+        assert_eq!(feature.links, feature_manifest.links.unwrap());
+        assert_eq!(feature.border_color, feature_manifest.border_color.unwrap());
+        assert_eq!(feature.tile_text_color, feature_manifest.tile_text_color.unwrap());
+    }
+
+    #[test]
+    fn map_feature_with_translations() {
+        let language_code = "de";
+        let links = HashMap::from([
+            ("some_link".to_string(), "https://example.de/1".to_string()),
+        ]);
+        let translation = DecodedFeatureManifest {
+            name: Some("testManifest_de".to_string()),
+            author: Some("testAuthor".to_string()),
+            version: Some("0.1.2".to_string()),
+            description: Some("testDescription_de".to_string()),
+            thumbnail: Some("assets/thumbnail.png".to_string()),
+            thumbnail_color: Some("#FFFFFF".to_string()),
+            primary_color: Some("#000000".to_string()),
+            border_color: Some("#000001".to_string()),
+            tile_text_color: Some("#000002".to_string()),
+            links: Some(links),
+            translations: None,
+        };
+        let feature_manifest = DecodedFeatureManifest {
+            links: Some(HashMap::from([
+                 ("some_link".to_string(), "https://some_link.com".to_string()),
+                 ("different_link".to_string(), "https://different_link.com".to_string()),
+            ])),
+            translations: Some(HashMap::from([
+                (language_code.to_string(), translation.clone()),
+            ])),
+            ..Default::default()
+        };
+        let features_dir = "features";
+        let feature_id = "feature_x";
+        let feature = map_feature(features_dir, feature_id, language_code, feature_manifest.clone()).unwrap();
+        assert_eq!(feature.path, format!("{}/{}", features_dir, feature_id));
+        assert_eq!(feature.id, feature_id.to_string());
+        assert_eq!(feature.name, translation.name.unwrap());
+        assert_eq!(feature.author, translation.author);
+        assert_eq!(feature.version, translation.version);
+        assert_eq!(feature.description, translation.description);
+        assert_eq!(feature.thumbnail, Some(format!("{}/{}/{}", features_dir, feature_id, translation.thumbnail.unwrap())));
+        assert_eq!(feature.thumbnail_color, translation.thumbnail_color.unwrap());
+        assert_eq!(feature.primary_color, translation.primary_color.unwrap());
+        assert_eq!(
+            feature.links, 
+            HashMap::from([
+                 ("some_link".to_string(), "https://example.de/1".to_string()),
+                 ("different_link".to_string(), "https://different_link.com".to_string()),
```

Comment:
Maybe test some stuff outside the happy path? A random string as URL? An url with `http`? What would happen in that case?

Reply:
URLs are not validated. It is assumed that FeatureManifest json is correctly defined.

Reply:
Again, maybe static checks for manifests?

Reply:
Could be, or we have a CLI to modify/create manifests.